sql/core/v2.3/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala

```diff
@@ -451,5 +452,31 @@ class OrcFilterSuite extends OrcTest with SharedSparkSession {
       ).get.toString
     }
   }
+
+  test("filter pushdown - local date") {
+    val dates = Seq("2017-08-18", "2017-08-19", "2017-08-20", "2017-08-21").map { day =>
+      LocalDate.parse(day)
+    }
+    withSQLConf(SQLConf.DATETIME_JAVA8API_ENABLED.key -> "true") {
```

Comment:
It makes me realize that this config can break data source v1/v2 as it changes the value in `Filter`.

Shall we fix `DataSourceAnalysis.translateLeafNodeFilter` instead to always return java 7 Date/Timestamp?

Reply:
@cloud-fan Here is the PR https://github.com/apache/spark/pull/28272

Reply:
@dongjoon-hyun ^^ The PR covers this PR

Reply:
Got it. Thanks, @MaxGekk !